src/oauth2.rs

```diff
@@ -120,14 +123,37 @@ pub fn dc_get_oauth2_access_token(
                     false,
                 )
             };
-        let mut token_url = replace_in_uri(&token_url, "$CLIENT_ID", oauth2.client_id);
-        token_url = replace_in_uri(&token_url, "$REDIRECT_URI", &redirect_uri);
-        token_url = replace_in_uri(&token_url, "$CODE", code.as_ref());
-        if let Some(ref token) = refresh_token {
-            token_url = replace_in_uri(&token_url, "$REFRESH_TOKEN", token);
+
+        // to allow easier specification of different configurations,
+        // token_url is in GET-method-format, sth. as https://domain?param1=val1&param2=val2 -
+        // convert this to POST-format ...
```

Comment:
Did  you consider using an external tool, e.g. the url or urlparse crate?

Reply:
i had a rough look, however, eg. https://docs.rs/url/2.1.0/url/ is too detailed, eg. we had to call url.scheme()+url.user()+url.password()+url.host()+url.port()+url.path()+someSeparators just to get post_url, does not look more readable to me.

also, maybe we should also _not_ use the GET-method-syntax, this seems to cause more confusion than helping. also, eg, in init-token, the strings MUST NOT be urlencoded, if needed, this is done by 

maybe we should use sth. more clear as https://domain.com?POST?client_id=$CLIENT_ID&name2=val2, no idea if there is a standard format to put a POST-request into a single string.